modules/foreman_organization.py

```diff
@@ -113,11 +106,14 @@ def main():
 
     module.connect()
 
-    entity = find_organization(module, name=entity_dict['name'], failsafe=True)
+    try:
+        entity = module.find_resource_by_name('organizations', name=entity_dict['name'], failsafe=True)
+    except Exception as e:
+        module.fail_json(msg='Failed to find entity: %s ' % e)
```

Comment:
Shouldn't `failsafe=True` handle this?

Reply:
I guess this catches other problems like could not find server.
Failsafe only returns None if the search result was empty.

Reply:
Sorry, I mean should `failsafe=True` handle this? Rather than doing something like this in every module?

Reply:
Good question. @evgeni you designed this pattern.

Reply:
and you did not yell at me when I posted the PR. ;)
I'll have a look/think about it.

Reply:
In your PR it was not yet replicated code...

Reply:
In a way it was, because we already had that pattern in many modules:
https://github.com/theforeman/foreman-ansible-modules/blob/c45f41c191165c7f0bc8682198dcf9b7fdaeeed8/modules/foreman_environment.py#L123-L126

I like your decorator idea!

Reply:
Also `failsafe` is something else, it's basically saying "get me 0 or 1" entities instead of "give me 1 entity". Bad naming I carried over from the nailgun backend.

Reply:
Since this looks like bigger refactoring, i moved it into a separate pr:
https://github.com/theforeman/foreman-ansible-modules/pull/281